cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java

```diff
@@ -156,6 +159,28 @@ public void deleteBefore(long timestamp) {
     // todo: implement metric ttl
   }
 
+  private void replaceTagsInListIfNeeded(List<TagValue> tagValues) {
```

Comment:
at some point we should use Map<String, String> instead of List<TagValue> I think... We'll see